src/core/net/ip6.cpp

```diff
@@ -1213,6 +1213,7 @@ Error Ip6::HandleDatagram(Message &aMessage, Netif *aNetif, const void *aLinkMes
         {
             // Remove encapsulating header and start over.
             aMessage.RemoveHeader(aMessage.GetOffset());
+            Get<MeshForwarder>().LogMessage(MeshForwarder::kMessageReceive, aMessage, nullptr, kErrorNone);
```

Comment:
Maybe not best structured to call MeshForwarder from here.
But fine to me for better logging.